src/FilmCollection/FilmCollection.js

```diff
@@ -2,22 +2,31 @@
 import React from 'react';
 import FilmAsset from '../FilmAsset/FilmAsset';
 import '../style.css';
+import { Route, Switch }  from 'react-router-dom';
 //constants
 const key = 'qwert1';
 
 const FilmCollection = (props) => {
 
-    const { films, filmsNumber, filmsSameGenre, isFilmClicked } = props;
+    const { films, filmsNumber, filmsSameGenre } = props;
 
-    const filmCollection = isFilmClicked ? filmsSameGenre : films;
+    const searchFilmCollection = (<ul className='film-collection' >
+        { films.map((film, i) => <FilmAsset key={key+i} film={film} /> )}
```

Comment:
It's a bad practice to use an array index for `key`(I mean the case when the list can be sorted). More info: https://reactjs.org/docs/lists-and-keys.html